text/0000-or-patterns.md

```diff
@@ -0,0 +1,666 @@
+would help, there are some where it would. Let's go through a few of them.
+
+[precursor]: https://github.com/rust-lang/rfcs/blob/de235887a80555427314c7eb25c6214523d50cce/text/0000-pipe-in-patterns.md
+
+1. One example which was raised in the [precursor] to this RFC was building a
+   state machine which is iterating through `chars_indices`:
+
+   ```rust
+   match iter.next() {
+       Some(_, ' ' | '\n' | '\r' | '\u{21A1}') => {
+           // Change state
+       }
+       Some(index, ch) => {
+           // Look at char
+       }
+       None => return Err(Eof),
+   }
+   ```
+
+[GHC proposal]: https://github.com/osa1/ghc-proposals/blob/or_patterns/proposals/0000-or-patterns.rst#real-world-examples
+
+2. Other examples are listed in the equivalent [GHC proposal].
+
+3. Another example which was provided in the [precursor] RFC was:
+
+   ```rust
+   for event in event_pump.poll_iter() {
+       use sdl2::event::Event;
+       use sdl2::keyboard::Keycode::{Escape, Q};
+       match event {
+           Event::KeyDown { keycode: Some(Escape | Q), ... } => break 'game,
+           _ => {},
+       }
+       ...
+   }
+   ```
+
+4. Other cases where this feature was requested include:
+   + <https://github.com/rust-lang/rust/issues/15219>
+   + <https://github.com/rust-lang/rust/issues/14516>
+
+[alercah_discord]: https://discordapp.com/channels/442252698964721669/448237931136679936/483325957130813440
+
+5. Another use case due to [@alercah][alercah_discord] is:
+
+   ```rust
+   pub fn is_green(self) -> bool {
+       match self {
+           | Tile::Suited(Suit::Souzu, 2 | 3 | 4 | 6 | 8)
+           | Tile::Dragon(Dragon::Green) => true,
+           _ => false,
+       }
+   }
+   ```
+
+6. Some further examples found with sourcegraph include:
+
+   From [cc-rs](https://sourcegraph.com/github.com/alexcrichton/cc-rs/-/blob/src/lib.rs#L1307:18):
+
+   ```rust
+    match (self.cpp_set_stdlib.as_ref(), cmd.family) {
+        (None, _) => {}
+        (Some(stdlib), ToolFamily::Gnu | ToolFamily::Clang) => {
+            cmd.push_cc_arg(format!("-stdlib=lib{}", stdlib).into());
+        }
+        _ => {
+            ...
+        }
+    }
+   ```
+
+   From [capnproto](https://sourcegraph.com/github.com/capnproto/capnproto-rust/-/blob/capnp/src/private/layout.rs#L1979-2002):
+
+   ```rust
+   // Check whether the size is compatible.
+   match expected_element_size {
+       None | Some(Void | InlineComposite) => (),
+       Some(Bit) => { ... }
+       Some(Byte | TwoBytes | FourBytes | EightBytes) => { ... },
+       ...
+   }
+   ```
+
+   From [chrono](https://sourcegraph.com/github.com/chronotope/chrono/-/blob/src/format/parsed.rs#L271-308):
+
+   ```rust
+   fn resolve_year(y: Option<i32>, q: Option<i32>,
+                   r: Option<i32>) -> ParseResult<Option<i32>> {
+       match (y, q, r) {
+           (y, None, None) => Ok(y),
+           (Some(y), q, r @ (Some(0...99) None)) => { ... },
```

Comment:
Missing `|` between the `Some` and `None` (I assume, Sourcegraph appears to not be Safari compatible so I can't view the original)